models/User.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require('mongoose')
+const bcrypt = require('bcryptjs')
+const jwt = require('jsonwebtoken')
+
+const UserSchema = new mongoose.Schema({
+    name: {
+        type: String, 
+        required: [true, 'Please provide name.'],
+        minLength: 3, 
+        maxLength: 50, 
+    }, 
+    email: {
+        type: String, 
+        required: [true, 'Please provide email.'],
+        match: [
+            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+        'Please provide valid email',
+        ],
+        unique: true,
+    }, 
+    password: {
+        type: String, 
+        required: [true, 'Please provide password.'],
+        minLength: 6, 
+    }, 
+})
+
+UserSchema.pre('save', async function (next) {
```

Comment:
```suggestion
UserSchema.pre('save', async function () {
```
Can get rid of this parameter as it's not used for anything.